src/scripts/main.js

```diff
@@ -1,5 +1,30 @@
 'use strict';
 
+const spider = document.getElementsByClassName('spider')[0];
+const wall = document.getElementsByClassName('wall')[0];
+
 document.addEventListener('click', (e) => {
-  // write code here
+  if (e.target.closest('.spider')) {
+    return;
+  }
+
+  if (e.target.closest('.wall')) {
+    const spiderCentered = spider.clientWidth / 2;
```

Comment:
Consider checking if 'spider.clientWidth' is defined to avoid potential runtime errors if the 'spider' element is not found or not yet rendered.